src/schema/sources.ts

```diff
@@ -1483,21 +1495,36 @@ export const resolvers: IResolvers<any, Context> = {
         inputHandle !== current.handle,
       );
 
+      if (!isNullOrUndefined(isPrivate) && current.private !== isPrivate) {
+        const approved = await ctx.con
+          .getRepository(SquadPublicRequest)
+          .findOne({
+            where: { sourceId, status: SquadPublicRequestStatus.Approved },
+          });
+
+        if (!approved) {
+          throw new ValidationError(SourceRequestErrorMessage.SQUAD_INELIGIBLE);
+        }
+      }
+
       try {
         const editedSourceId = await ctx.con.transaction(
           async (entityManager) => {
             const repo = entityManager.getRepository(SquadSource);
+            const updates: Partial<SquadSource> = {
+              name,
+              handle,
+              description,
+              memberPostingRank: sourceRoleRank[memberPostingRole],
+              memberInviteRank: sourceRoleRank[memberInviteRole],
+            };
+
+            if (!isNullOrUndefined(isPrivate)) {
+              updates.private = isPrivate;
+            }
```

Comment:
Maybe streamline even more so we set it only if private changed and in the same place as approved check? https://github.com/dailydotdev/daily-api/commit/e997d8567da8c205d1dd31d2ed1d538dd6f0b73f

Reply:
Right. Yeah, we can do it within the previous if statement. We might have to bring the whole `updates` object way above, but shouldn't hurt.